src/concrete_intersection.jl

```diff
@@ -100,6 +100,109 @@ function intersection(H1::AbstractHyperrectangle{N},
     return Hyperrectangle(high=high, low=low)
 end
 
+"""
+    intersection(P1::AbstractHPolygon{N},
+                 P2::AbstractHPolygon{N}
+                )::Union{HPolygon{N}, EmptySet{N}} where N<:Real
+
+Return the intersection of two polygons in constraint representation.
+
+### Input
+
+- `P1` -- first polygon
+- `P2` -- second polygon
+
+### Output
+
+If the polygons do not intersect, the result is the empty set.
+Otherwise the result is the polygon that describes the intersection.
+
+### Algorithm
+
+We just combine the constraints of both polygons.
```

Comment:
Maybe extend a bit: "The thing is that adding constraints all at once gives a worst-case quadratic algorithm. Interleaved insertion is worst-case linear."